Monika After Story/game/script-islands-event.rpy

```diff
@@ -215,73 +178,130 @@ init -25 python in mas_island_event:
     # during composite image building
     # NOTE: Use functools.partial instead of renpy.partial because the latter has an argument conflict. Smh Tom
     # Islands
-    _IslandsImgDataHolder(
-        "isld_0",
+    IslandsImageDefination(
+        "island_0",
         default_unlocked=True,
         partial_disp=functools.partial(
             ParallaxSprite,
-            x=-64,
+            x=-85,
             y=660,
-            z=20,
+            z=15,
             on_click=True
         )
     )
-    _IslandsImgDataHolder(
-        "isld_1",
+    def __isld_1_transform_func(transform, st, at):
+        """
+        A function which we use as a transform, updates the child
+        """
+        amp = 0.02
+        frenq_1 = 1.0 / 9.0
+        frenq_2 = 1.0 / 3.0
+
+        transform.ypos = math.cos(at*frenq_1) * math.sin(at*frenq_2) * amp
```

Comment:
do you have any ideas on how this math works? (also the other transform functions)

Reply:
I think so? I just played around with the trigonometric funcs a little until the movement looked decent. I thought about leaving some comments, but wasn't sure what exactly to write. It's more math than programming, and explaining trigonometry didn't feel right.

Reply:
>Or did you mean the use of at, frenq, amp, and other variables to control the function?

Sort of this. Main point should be a comment to explain what the result gives in relation to `at`. For example, as `at` increases, what happens to the result value. Does the result value trace a particular path? Additional comments on how the `frenq` and `amp` vars affect the result as well.

Reply:
I tried to make it more verbose and added a comment explaining the logic behind it. Let me know if this is any better now.

Reply:
bit more verbose then what I was going for. I ended up graphing each one and added images to the dev folder.

Reply:
Technically I could make islands draw their path as they move while in dev mode.

Reply:
that would be cool, but id make it togglable.